src/core/webview/webviewMessageHandler.ts

```diff
@@ -37,33 +37,38 @@ import { openMention } from "../mentions"
 import { telemetryService } from "../../services/telemetry/TelemetryService"
 import { TelemetrySetting } from "../../shared/TelemetrySetting"
 import { getWorkspacePath } from "../../utils/path"
-import { Mode, PromptComponent, defaultModeSlug, getModeBySlug, getGroupName } from "../../shared/modes"
+import { Mode, defaultModeSlug, getModeBySlug, getGroupName } from "../../shared/modes"
 import { getDiffStrategy } from "../diff/DiffStrategy"
 import { SYSTEM_PROMPT } from "../prompts/system"
 import { buildApiHandler } from "../../api"
+import { GlobalState } from "../../schemas"
 
 export const webviewMessageHandler = async (provider: ClineProvider, message: WebviewMessage) => {
+	// Utility functions provided for concise get/update of global state via contextProxy API.
```

Comment:
provider.get/updateGlobalState is marked as deprecated in ClineProvider.ts, so don't use it.

`getGobalState` is more concise than `provider.contextProxy.getValue` (and used a lot throughout this file, so worthwhile to offer a more concise syntax).